stackslib/src/chainstate/coordinator/mod.rs

```diff
@@ -723,7 +739,30 @@ pub fn get_reward_cycle_info<U: RewardSetProvider>(
         }
     } else {
         Ok(None)
+    };
+
+    if let Ok(Some(reward_cycle_info)) = reward_cycle_info.as_ref() {
+        // cache the reward cycle info as of the first sortition in the prepare phase, so that
+        // the Nakamoto epoch can go find it later
+        let ic = sort_db.index_handle(sortition_tip);
+        let prev_reward_cycle = burnchain
+            .block_height_to_reward_cycle(burn_height)
+            .expect("FATAL: no reward cycle for burn height");
+
+        if prev_reward_cycle > 1 {
+            let prev_reward_cycle_start = burnchain.reward_cycle_to_block_height(prev_reward_cycle - 1);
```

Comment:
I think the prepare phase start math here should go in the `burnchain/mod.rs` `PoxConstants` implementation. It would also be good to dedupe with the `static_is_in_prepare_phase`.

Reply:
Happy to put this into `PoxConstants` but the act of deduping with `static_is_in_prepare_phase` will entail writing even more code than leaving it as-is (we'd need a `prepare_phase_start`, `static_reward_cycle_to_block_height`, and `static_prepare_phase_start`)